### DISCUSSION TITLE
Can I re-run an azd template on an existing templated codebase to update it?

### DISCUSSION THREAD

**Discussion by TWolverson** [Azure/azure-dev#1224] [Q&A]
The azd standard templates have changed a fair bit during development - this is expected, and not in itself a problem. Suppose I have initialised from a solution template a couple of versions ago, made some customisations, and now I want to automatically update my solution to match the present version's conventions so that my existing project and any future projects I start with azd are consistent. Is there a defined approach to doing that? Running `azd init` again errors:
`Error: environment already initialized to test`

which makes sense.

If I could pass an option that tries to merge the current version of the template into my (source-controlled) actual solution, ensuring changes are staged, that could be workable.

**Comment by aaronpowell** [comment]
To a degree, this overlaps with the feature I proposed here: https://github.com/Azure/azure-dev/issues/1056
